### PR TITLE
[PR Template] Add codeblock to capture deprecation note associated with PR

### PR DESCRIPTION
**Pull request by paulbouwer**
**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:

Adds a code block to the PR Template for the **kubernetes/kubernetes** repo to capture the **deprecation note** information as requested in [Issue kubernetes/sig-release#668](https://github.com/kubernetes/sig-release/issues/668)

**Which issue(s) this PR fixes**:

Refers to Issue [kubernetes/sig-release#668](https://github.com/kubernetes/sig-release/issues/668)

**Special notes for your reviewer**:

This is a smaller focused PR that was extracted from [PR kubernetes/kubernetes #81278](https://github.com/kubernetes/kubernetes/pull/81278)

This will require an update to the [release notes tool](https://github.com/kubernetes/release/tree/master/cmd/release-notes) for the next release cycle (1.17) to ensure that the **deprecation note** information is parsed and processed.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:
```docs

```

/cc @saschagrunert @onyiny-ang
/sig release

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (saschagrunert) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
Indicate whether this PR implements a change that requires users to be informed of a deprecation of an API.
```

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (saschagrunert) [inline .github/PULL_REQUEST_TEMPLATE.md]: I would leave this empty to keep consistency with the other blocks:

```suggestion
```

**Review by saschagrunert — CHANGES_REQUESTED** [review]
@paulbouwer thanks, I have two comments.

**Comment by saschagrunert** [comment]
@paulbouwer are you still around to work on this or should I take it over? 
(I'll assume that the overall direction will change a bit)

**Comment by justaugustus** [comment]
I've mentioned on a few SIG Release calls, so adding here as well...
I think the right path forward might be to introduce a new label, `kind/deprecation`, which when applied along with a release note, our release notes tooling will treat as a deprecation note.

I've opened a PR to discuss and make that change here: https://github.com/kubernetes/test-infra/pull/15040

Let's see where that discussion leads before moving on this.
/hold
/priority important-longterm
/kind feature
/milestone v1.17

**Comment by saschagrunert** [comment]
Thanks @justaugustus, do we want to do the same for the primary SIG? (https://github.com/kubernetes/kubernetes/pull/82449)

**Comment by josiahbjorgaard** [comment]
@justaugustus Will this be making it into v1.17 before code freeze this Thursday or can we change the milestone?

**Comment by justaugustus** [comment]
Consensus seems to be leaning towards the `kind/deprecation` label, so I'm going to close this PR.
We can come back to any template updates once that lands in https://github.com/kubernetes/test-infra/pull/15040.

/close

**Pull request closed without merging**
